### PR TITLE
Refactor/rooted path pathlib interface

### PR DESCRIPTION
**Pull request by filipovskyj02**
## Summary
- Add `__truediv__` (`/` operator), read-only properties (`name`, `stem`,
  `suffix`) and query methods (`exists`, `is_dir`, `is_file`,
  `is_relative_to`, `relative_to`, `as_posix`) to `RootedPath`
- Migrate existing callsites to use the new interface

### CODE REVIEW

File: hermeto/core/resolver.py [hunk 1]
```diff
@@ -66,7 +66,7 @@ def resolve_packages(request: Request) -> RequestOutput:
         for project_file in output.build_config.project_files:
             try:
                 subpath = project_file.abspath.relative_to(source_backup)
-                project_file.abspath = original_source_dir / subpath
+                project_file.abspath = (original_source_dir / subpath).path
```
Comment (tnevrlka) [inline hermeto/core/resolver.py:69]: It looks like this is not related to the exposure of `pathlib.Path` operations.
Should this be in another commit?

File: hermeto/core/rooted_path.py [hunk 2]
```diff
@@ -88,6 +91,49 @@ def __repr__(self) -> str:
...
+        return self._path.stem
+
+    @property
+    def suffix(self) -> str:
+        """Return the file extension of the final component."""
+        return self._path.suffix
+
+    def exists(self) -> bool:
+        """Return whether the path points to an existing filesystem entry."""
+        return self._path.exists()
```
Comment (tnevrlka) [inline hermeto/core/rooted_path.py:119]: How were the exposed operations chosen? I assume read-only that cannot escape root? Might be worth describing.

File: hermeto/core/rooted_path.py [hunk 3]
```diff
@@ -88,6 +91,49 @@ def __repr__(self) -> str:
     def __hash__(self) -> int:
         return hash((self._path, self._root))
 
+    def __truediv__(self: RootedPathT, other: StrPath) -> RootedPathT:
+        """Join via ``join_within_root``; raises PathOutsideRoot on escape."""
+        return self.join_within_root(other)
```
Comment (tnevrlka) [inline hermeto/core/rooted_path.py:96]: Since `__truediv__` is the only operation behaving differently from `pathlib.Path`, would it make sense to have it in its own commit?

File: tests/unit/test_rooted_path.py [hunk 4]
```diff
@@ -123,6 +123,85 @@ def test_rooted_path_eq() -> None:
...
+    rp = RootedPath(test_path).join_within_root(path_suffix)
+    assert getattr(rp, prop) == expected
+
+
+def test_name_at_root(test_path: Path) -> None:
+    rp = RootedPath(test_path)
+    assert rp.name == test_path.name
+
+
+
```
Comment (tnevrlka) [inline tests/unit/test_rooted_path.py]: nit: looks like three newlines

File: tests/unit/test_rooted_path.py [hunk 5]
```diff
@@ -123,6 +123,85 @@ def test_rooted_path_eq() -> None:
...
+@pytest.mark.parametrize(
+    "prop, path_suffix, expected",
+    [
+        pytest.param("name", "subpath", "subpath", id="name"),
+        pytest.param("name", "archive.tar.gz", "archive.tar.gz", id="name-with-dots"),
+        pytest.param("stem", "archive.tar", "archive", id="stem"),
+        pytest.param("suffix", "archive.tar", ".tar", id="suffix"),
+        pytest.param("suffix", "no_extension", "", id="suffix-empty"),
+    ],
+)
```
Comment (tnevrlka) [inline tests/unit/test_rooted_path.py:163]: Is it possible to simplify the tests? 

Exposed operations (besides `__truediv__`) are supposed to call `pathlib.Path`'s operations and should thus behave 1:1.

I don't think it's necessary to try to test edge cases, I would assume that just checking the happy case is behaving same as `pathlib.Path` should be enough

**Review by tnevrlka — COMMENTED** [review]
Overall looks good to me, few comments for thought

File: hermeto/core/rooted_path.py (hunk 2 above)
Comment (filipovskyj02) [reply hermeto/core/rooted_path.py:119]: Yes, as the minimum issue desc stated, wanted to open the pr for feedback early, so open to expanding on it.

Comment (filipovskyj02) [reply hermeto/core/rooted_path.py:119]: Added a comment for it for now

File: hermeto/core/resolver.py (hunk 1 above)
Comment (filipovskyj02) [reply hermeto/core/resolver.py:69]: Moved to the migration commit

File: hermeto/core/rooted_path.py (hunk 3 above)
Comment (filipovskyj02) [reply hermeto/core/rooted_path.py:96]: Added a third commit for it

File: tests/unit/test_rooted_path.py (hunk 5 above)
Comment (a-ovchinnikov) [reply tests/unit/test_rooted_path.py:163]: Agreed, these tests are an overkill for all cases when the underlying implementation is just passed through.

Comment (filipovskyj02) [reply tests/unit/test_rooted_path.py:163]: Simplified

File: hermeto/core/rooted_path.py (hunk 2 above)
Comment (taylormadore) [reply hermeto/core/rooted_path.py:119]: What about I/O methods? Things like open, read_text, write_text, mkdir, etc.
